test/e2e/redundancy/gateway_failover.go

```diff
@@ -7,42 +7,26 @@ import (
 	. "github.com/onsi/gomega"
 	"github.com/submariner-io/submariner/test/e2e/dataplane"
 	"github.com/submariner-io/submariner/test/e2e/framework"
-	appsv1 "k8s.io/api/apps/v1"
 )
 
 var _ = Describe("[redundancy] Gateway fail-over tests", func() {
 	f := framework.NewDefaultFramework("gateway-redundancy")
 
 	When("one gateway node is configured and the submariner engine pod fails", func() {
 		It("should start a new submariner engine pod and be able to connect from another cluster", func() {
-			testOneGatewayNode(f)
+			testEnginePodRestartScenario(f)
 		})
 	})
 
-	When("two gateway nodes are configured with one submariner engine replica and the gateway node fails", func() {
-		It("should start a new submariner engine pod on the second gateway node and be able to connect from another cluster", func() {
-			testTwoGatewayNodesWithOneReplica(f)
+	When("a new node is labelled as gateway node and label on existing gateway node is removed", func() {
+		It("should start a submariner engine on the second gateway node and be able to connect from another cluster", func() {
+			testGatewayFailOverScenario(f)
 		})
 	})
 
-	When("two gateway nodes are configured with two submariner engine replicas and the active gateway node fails", func() {
-		var deployment *appsv1.Deployment
-
-		BeforeEach(func() {
-			deployment = f.FindSubmarinerEngineDeployment(framework.ClusterA)
```

Comment:
`FindSubmarinerEngineDeployment` function can be removed.

Reply:
Done